### PR TITLE
[Core][Originator] Base on promotion codes instead of originator

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets |
| License         | MIT

As ``Promotion``s have now unique codes, there is no need to use separate service to determine origin of promotion (or any other adjustment).

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
Looks good! 👍 

Since we have adjustments applied on different levels, what do you think about adding one more scenario for item promotion?

File: src/Sylius/Component/Order/Model/AdjustmentInterface.php [hunk 1]
```diff
@@ -93,6 +92,16 @@ public function isCredit();
      */
     public function isLocked();
 
+    /**
+     * @return int
```
Comment (lchrusciel) [inline src/Sylius/Component/Order/Model/AdjustmentInterface.php]: According to spec it should be string instead of int

**Comment by Zales0123** [comment]
@michalmarcinkowski ok, I fact there is no possibility for such scenario 😄  we have now two types of unit promotions - fixed and percentage and both uses ``price range`` and ``taxon`` filter. Therefore, to revert promotion we should modify specific product taxon or price (!), which I think is not the point in this case ;)

**Comment by pjedrzejewski** [comment]
🎆

**Comment by michalmarcinkowski** [comment]
@Zales0123 there is possibility! 😄  We can use order promotion rule together with item promotion action, e.g.: `Given the promotion gives 5% off on every product to every order with items total at least "€80.00"`

**Comment by dantleech** [comment]
This change couples adjustments to promotions? What if an adjustment is made by a non-promotion?

**Comment by dantleech** [comment]
Hmm. Now I wonder if I should be using promotions ...

**Comment by michalmarcinkowski** [comment]
It doesn't couple adjustments with promotions. Promotions just use `originCode` field from adjustment to be able to recognize them. Non promotion adjustments are not affected by this change (we already have also e.g. tax adjustments).

**Comment by dantleech** [comment]
ok, yes I see now. so I can still reference something with the combination of the `type` and `origin_code`. Thanks :)

**Pull request merged**